test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CSharp7.cshtml

```diff
@@ -0,0 +1,32 @@
+@{
+    var nameLookup = new Dictionary<string, (string FirstName, string LastName, object Extra)>()
+    {
+        ["John Doe"] = ("John", "Doe", true)
+    };
+
+    int Sixteen = 0b0001_0000;
+    long BillionsAndBillions = 100_000_000_000;
```

Comment:
Looks like any complex double of any kind has never worked in implicit expression. Will add tests for doing this with explicit expressiong.

Bug for implicit expressions: https://github.com/aspnet/Razor/issues/1218